src/main/java/org/elasticsearch/plugins/PluginManager.java

```diff
@@ -19,674 +19,557 @@
-                            Files.setPosixFilePermissions(file, perms);
+            if (terminal.verbosity().enabled(NORMAL)) {
+                boolean required = false;
+
+                // Iterates over all installed plugins to check if the plugin to uninstall is required by another plugin
+                for (PluginDescriptor plugin : local.list()) {
+                    Set<PluginDescriptor> dependencies = plugin.dependencies();
+                    if (dependencies != null) {
+                        for (PluginDescriptor dependency : dependencies) {
+                            if (StringUtils.equals(removable.name(), dependency.name())) {
+
+                                if (!required) {
+                                    required = true;
+                                    terminal.println("The following plugin(s) depends on the plugin about to be removed:");
+                                }
+                                terminal.println("\t %s -> %s", plugin, dependency);
+                            }
                         }
-                        return FileVisitResult.CONTINUE;
                     }
-                });
-            } else {
-                debug("Skipping posix permissions - filestore doesn't support posix permission");
-            }
-            debug("Installed " + name + " into " + toLocation.toAbsolutePath());
-            potentialSitePlugin = false;
-        }
-
-        Path configFile = extractLocation.resolve("config");
-        if (Files.isDirectory(configFile)) {
-            Path configDestLocation = pluginHandle.configDir(environment);
-            debug("Found config, moving to " + configDestLocation.toAbsolutePath());
-            moveFilesWithoutOverwriting(configFile, configDestLocation, ".new");
-            debug("Installed " + name + " into " + configDestLocation.toAbsolutePath());
-            potentialSitePlugin = false;
-        }
+                }
 
-        // try and identify the plugin type, see if it has no .class or .jar files in it
-        // so its probably a _site, and it it does not have a _site in it, move everything to _site
-        if (!Files.exists(extractLocation.resolve("_site"))) {
-            if (potentialSitePlugin && !FileSystemUtils.hasExtensions(extractLocation, ".class", ".jar")) {
-                log("Identified as a _site plugin, moving to _site structure ...");
-                Path site = extractLocation.resolve("_site");
-                Path tmpLocation = environment.pluginsFile().resolve(extractLocation.getFileName() + ".tmp");
-                Files.move(extractLocation, tmpLocation);
-                Files.createDirectories(extractLocation);
-                Files.move(tmpLocation, site);
-                debug("Installed " + name + " into " + site.toAbsolutePath());
+                if (required) {
+                    String confirm = terminal.readText("Confirm removal of plugin %s? [yes/no]", removable.name());
```

Comment:
Also may be this should be documented in the asciidoc page with something like that `if another plugin depends to the plugin you are removing, you'll need to confirm that you actually want to remove it.`

Which leads me to another question: is there any way to force the plugin removal? I'm thinking about automatic scripts.